examples/own_dataset/train.py

```diff
@@ -0,0 +1,209 @@
+from chainer_chemistry.datasets import NumpyTupleDataset
+
+from rdkit import Chem
+
+
+class GraphConvPredictor(chainer.Chain):
+
+    def __init__(self, graph_conv, mlp=None):
+        """Initialize GraphConvPredictor
+
+        Args:
+            graph_conv: graph convolution network to obtain molecule feature
+                        representation
+            mlp: multi layer perceptron, used as final connected layer.
+                It can be `None` if no operation is necessary after
+                `graph_conv` calculation.
+        """
+
+        super(GraphConvPredictor, self).__init__()
+        with self.init_scope():
+            self.graph_conv = graph_conv
+            if isinstance(mlp, chainer.Link):
+                self.mlp = mlp
+        if not isinstance(mlp, chainer.Link):
+            self.mlp = mlp
+
+    def __call__(self, atoms, adjs):
+        x = self.graph_conv(atoms, adjs)
+        if self.mlp:
+            x = self.mlp(x)
+        return x
+
+
+def main():
+    # Supported preprocessing/network list
+    method_list = ['nfp', 'ggnn', 'schnet', 'weavenet', 'rsgcn']
+    scale_list = ['standardize', 'none']
+
+    parser = argparse.ArgumentParser(
+        description='Regression with own dataset.')
+    parser.add_argument('datafile', type=str)
+    parser.add_argument('--method', '-m', type=str, choices=method_list,
+                        default='nfp')
+    parser.add_argument('--label', '-l', nargs='+',
+                        help='target label for regression')
+    parser.add_argument('--scale', type=str, choices=scale_list,
+                        default='standardize', help='Label scaling method')
+    parser.add_argument('--conv-layers', '-c', type=int, default=4)
+    parser.add_argument('--batchsize', '-b', type=int, default=32)
+    parser.add_argument('--gpu', '-g', type=int, default=-1)
+    parser.add_argument('--out', '-o', type=str, default='result')
+    parser.add_argument('--epoch', '-e', type=int, default=20)
+    parser.add_argument('--unit-num', '-u', type=int, default=16)
+    parser.add_argument('--seed', '-s', type=int, default=777)
+    parser.add_argument('--train-data-ratio', '-t', type=float, default=0.7)
+    args = parser.parse_args()
+
+    seed = args.seed
+    train_data_ratio = args.train_data_ratio
+    method = args.method
+    if args.label:
+        labels = args.label
+        class_num = len(labels) if isinstance(labels, list) else 1
+    else:
+        sys.exit("Error: No target label is specified.")
+
+    # Dataset preparation
+    dataset = None
+
+    # Postprocess is required for regression task
+    def postprocess_label(label_list):
+        return numpy.asarray(label_list, dtype=numpy.float32)
+
+    print('preprocessing dataset...')
+    preprocessor = preprocess_method_dict[method]()
+    parser = CSVFileParser(preprocessor,
+                           postprocess_label=postprocess_label,
+                           labels=labels, smiles_col='SMILES')
```

Comment:
Is it intentional that you hard-coded the value of `smiles_col`?

Reply:
Yes, I want to show that we can specify the column name by `smiles_col` in this example.